src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -513,14 +512,18 @@ public String toString()
 
             });
           }
-        }
-        catch (IOException ie) {
-          l.handleException(ie, DefaultEventLoop.this);
+        } catch (UnresolvedAddressException e) {
+          logger.error("Inet Address {} is not resolvable.", address, e);
```

Comment:
Will it be okay to log it as debug and turn on the debug logging for com.datatorrent.netlet in log4j.properties in the production application?

-
Chetan

From: Vlad Rozov <notifications@github.com<mailto:notifications@github.com>>
Reply-To: DataTorrent/Netlet <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, March 30, 2016 at 6:45 PM
To: DataTorrent/Netlet <Netlet@noreply.github.com<mailto:Netlet@noreply.github.com>>
Cc: Chetan Narsude <chetan@apache.org<mailto:chetan@apache.org>>
Subject: Re: [DataTorrent/Netlet] Added handling of unresolvable host names (#42)


In src/main/java/com/datatorrent/netlet/DefaultEventLoop.java<https://github.com/DataTorrent/Netlet/pull/42#discussion_r57991434>:

> @@ -513,14 +512,18 @@ public String toString()
>
>              });
>            }
> -        }
> -        catch (IOException ie) {
> -          l.handleException(ie, DefaultEventLoop.this);
> +        } catch (UnresolvedAddressException e) {
> +          logger.error("Inet Address {} is not resolvable.", address, e);


Will it be OK to log this as the error as we need it for diagnostic in production deployment? We need the exception to be logged and if it is not logged by netlet, it will not be logged or handled anywhere outside of AbstractClient. Another approach will be to change AbstractClient.handleException() to log using error level. Netlet consumers may overwrite handleException() and log or handle exception the way they need.

-
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub<https://github.com/DataTorrent/Netlet/pull/42/files/ba23e56aaa484d0beb234058d2e6a8651113697e#r57991434>